include/deal.II/grid/reference_cell.h

```diff
@@ -617,7 +617,7 @@ class ReferenceCell
    *   these orientation indices.
    */
   template <typename T>
-  unsigned int
+  unsigned char
```

Comment:
I thought about this (and the corresponding next change in this patch) when I introduced these replacement function. I didn't quite know what the right approach is.

At its core, the function returns (and the next function) takes an integer index. We generally use `unsigned int` for these, simply because that is the easiest way to represent "this is an index or count", even though in many cases the data type is strictly speaking larger than what is necessary to represent the index. We usually don't use `unsigned char` for a variety of reasons among which are that these numbers are difficult to print on the screen (because they are printed as characters rather than integers), and we have usually not cared about the space savings unless we keep big arrays for them. If we did care about space savings, I would think that `std::uint8_t` might be a better way to indicate "this is an index" than `unsigned char`. I'm curious to hear what you think.

Reply:
Good question - the short answer is that 'we use `unsigned char` for the orientations elsewhere'. I think the consistency in using `unsigned char` everywhere for orientation is a plus (we have enough other issues with inconsistent orientation definitions) though it being non-printing is annoying.

There are a couple of places where this data is stored in big arrays - we store `6 * n_cells` face orientation flags. Using the smaller size saves a lot of memory (if we used `std::bitset<3>` and my favorite heart mesh the high-water memory consumption mark would be about 200-300 MB higher - `8 * 6 * 2.4e6 / 1024 / 1024 = 109.86ish`).

Reply:
We could introduce our own type
```cpp
class Orientation
{
public:
   bool orientation() {return get_bit(flags, 0);}

  // etc
private:
   unsigned char flags;
};
```
or something similar with a way to easily convert it into an array index. There's a certain simplicity to using just a plain `unsigned char` here: at some point we need to hit bedrock and use an existing byte type.

It looks like `std::uint8_t` is the same as `unsigned char`: if I print them I get ASCII characters.

Reply:
Yes, it aggrieves me that the C++ standard makers have allowed `std::uint8_t` to be a typedef of `unsigned char`, making the integer unprintable :-(